server/web.js

```diff
@@ -314,3 +381,427 @@ function getPluginVersion(callback) {
+          if (mapValue) {
+            // Then we got a regex match
+
+            // Its actual indexInLine from the start of the entire line
+            //   depends on where our substring started (indexInLine)
+            mapValue["indexInLine"] = indexInLine + mapValue["indexInLine"];
+            // Start the next substring from where this regex ends
+            indexInLine = mapValue["indexInLine"] + mapValue["length"];
+
+            if (regexesMap.has(lineNumber)) {
+              regexesMap.set(lineNumber, regexesMap.get(lineNumber).concat([mapValue]));
+            } else {
+              regexesMap.set(lineNumber, [mapValue]);
+            }  
+          } else {
+            // If no mapValue, there are no more regexes on this line
+            // so we're done with the while loop
+            break;
+          }
+
+
+        }
+
+        
+      }
+    });
+  }
+
+  return regexesMap;
+}
+
+/** Get the location within a line and length of a regex match,
+ * given the result of a tre-agrep call */
+function getRegexLocationAndLength(stdout) {
+  if (!stdout) {
+    return;
+  }
+
+  // stdout returns ASCII numbers, so convert them to strings
+  var resultString = stdout.toString('utf8');
+
+  var values = resultString.split(':');
+  if (values.length < 3) {
+    // Not a legitimate match
+    return;
+  }
+
+  var lineNumber = parseInt(values[0]);
+  var relevantChars = values[1];
+  var indices = relevantChars.split('-');
+  var indexInLine = parseInt(indices[0]);
+  // Note: If *, only includes the len of things before the *
+  //   haven't tested if you have abc*xyz as the regex yet
+  var lengthToHighlight = parseInt(indices[1]) - parseInt(indices[0]);
+
+  var mapValue = {
+    'indexInLine': indexInLine,
+    'length': lengthToHighlight
+  };
+
+  return mapValue;
+}
+
+
+function getChunkedDiffs(ops, threshold, baseline) {
+    if (!threshold) {
+      threshold = 10000;
+    }
+    // TODO: Very large threshold => no results
+
+    // If there have been no changes to the document,
+    // ops = {v:0}
+    if (!Array.isArray(ops)) {
+      return [];
+    }
+
+    var chunkedDiffs = [];
+
+    /* Setup the baseline of the document */ 
+    var firstOp = ops[0];
+
+    // The baseline for the next diff
+    var currentBaseline = {v:0};
+    sharedb.ot.apply(currentBaseline, firstOp);
+    // The doc to apply ops to
+    var currentDoc = {v:0};
+    sharedb.ot.apply(currentDoc, firstOp);
+
+    var lastTs = firstOp.m.ts;
+
+    // Create a diff for the first part, so that
+    // we can track original code
+    var baseDiff = diff.diffLines(baseline.trim(), currentBaseline.data.text.trim());
+    baseDiff.forEach(function(part) {
+      // Note: should only be one part
+      part.original = true;
+    });
+
+    chunkedDiffs.push(baseDiff);
+
+    /* Apply each op, and calculate a diff if two 
+       consecutive ops are far enough apart */
+    for (var i = 1; i < ops.length; i++) {
```

Comment:
Iterate without using a counter?

Reply:
Either iterate over a slice from 1, or: the 0th op immediately triggers a diff that performs whatever function the piece of code above is doing w.r.t. baseline.

Reply:
done.